documentation.md

```diff
@@ -0,0 +1,64 @@
+#Observer Pattern Implementation in Java
```

Comment:
The formatting for this README.md file is mostly incorrect. Please open the file in VS Code and observe the file via the Preview Pane to see the issues:
- Headings are misapplied
- Paragraph breaks are missing
- Proper in-line code formatting has not been used

Please fix these issues and update the PR.